plugins/test/chromatic_correction_test.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright © 2024 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import unittest
+
+import numpy
+
+from odemis import model
+from odemis.dataio.tiff import read_data
+from odemis.util.spot import MaximaFind
+from odemis.util.transform import AffineTransform, alt_transformation_matrix_to_implicit, _rotation_matrix_from_angle
+from plugins.chromatic_correction import (BEAD_QUANTITY, TOLERANCE,
+                                          find_corresponding_points, get_chromatic_correction_dict)
+
+PIXEL_SIZE = (1e-6, 1e-6)  # m
+# Default values when there the datasets are similar
+DEFAULT_SHEAR = 0
+DEFAULT_SCALE = [1, 1]
+DEFAULT_ROTATION = 0
+DEFAULT_TRANSLATION = [0, 0]
+
+
+class TestChromaticAberrationCorrection(unittest.TestCase):
+    """
+    Test the Chromatic Aberration Correction functions used in the plugin
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.das = read_data('FOV1_green.tif')
+
+    def test_chromatic_correction_with_fake_data(self):
+        """Given two images in which one is reference image and other is affine transformed image of reference image.
+         The calculated affine parameters should correct the transformed image parameters back to the reference image"""
+        # reference image
+        ref = self.das[0]
+        ref.metadata.update({model.MD_ROTATION: 0})
+        ref.metadata.update({model.MD_PIXEL_SIZE: PIXEL_SIZE})
+        ref.metadata.update({model.MD_SHEAR: 0})
+        ref.metadata.update({model.MD_POS: [0, 0]})
+        # Affine parameters that are used on reference image
+        # Translation can be any value. Scale is fixed to one if there is some rotation and shear else both scale, shear
+        # can be changed by keeping the rotation to zero
+        # Set 1: change scale and sheer, keeping the rotation unchanged
+        # Set 2: change the rotation and sheer, keeping the scale unchanged
+        affine_sets = {"set_1": {"shear": 0.9, "scale": [2, 3], "rotation": numpy.radians(0), "translation": [2, 1]},
+                       "set_2": {"shear": 0.4, "scale": [1, 1], "rotation": numpy.radians(5), "translation": [2, 3]}}
+
+        for affine_set in affine_sets.values():
+            SHEAR = affine_set["shear"]
+            SCALE = affine_set["scale"]
+            ROTATION = affine_set["rotation"]
+            TRANSLATION = affine_set["translation"]
```

Comment:
is there a reason to use all capital letters here and not `shear, scale, rotation, translation`?